clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -496,9 +504,21 @@ public void testCreate() throws ApiException, IOException {
         mockNonExistingPath(new ObjectLocation("lakefs", "repo", "main", "sub1/sub2/create.me"));
 
         StagingLocation stagingLocation = new StagingLocation().token("foo").physicalAddress(s3Url("/repo-base/create"));
-        when(stagingApi.getPhysicalAddress("repo", "main", "sub1/sub2/create.me", false))
+        when(stagingApi.getPhysicalAddress("repo", "main", "sub1/sub2/create.me"))
+                .thenReturn(getPhysicalAddressRequest);
+        when(getPhysicalAddressRequest.execute())
                 .thenReturn(stagingLocation);
 
+        when(stagingApi.linkPhysicalAddress(eq("repo"), eq("main"), eq("sub1/sub2/create.me"), any(StagingMetadata.class)))
+                .thenReturn(linkPhysicalAddressRequest);
```

Comment:
If you do this, why do you still verify at l. 532?  I think it _used to_ work because of some Mockito weirdness here where it's configured to allow _any_ calls (and return a "smart" `null`); now you need to mock it, but that probably means you don't have to verify.

Reply:
@arielshaqed You're not wrong, but the difference IMHO is in the feedback you get. If you use the capture in `when`, you'll get an error that no value was captured, so it looks like the test method is broken while actually the tested method might be broken. That's at least how I understand it.